52259tan/UofTCourseScheduleOptimizer
# Add User Input View and Demo

Hi everyone it's Percy here! 

I have recently made a view (and that include the View, Controller, Input Boundary, Input Data) where user can input the courses they want to take. Note: the max is 7. 

**Here's how to test the View:** 

1. Open the Main file in the Demo package and run it 
2. Type in the courses. Note: you have to use the correct format (eg. "_**ECO317H1 -F**_"). For more courses you can test and their format, go to the availableCourses.txt file. 
3. Press submit, confirm that you have something like this indicating the test is successful 

<img width="725" alt="Screen Shot 2023-11-19 at 3 16 34 PM" src="https://github.com/52259tan/UofTCourseScheduleOptimizer/assets/144645042/c158d3b9-3cec-4aa3-8b11-69975673f73e">

**But what if there's an error:** 
If you get an error like this: 
<img width="1030" alt="Screen Shot 2023-11-19 at 3 21 25 PM" src="https://github.com/52259tan/UofTCourseScheduleOptimizer/assets/144645042/16e81fdf-16c7-4d56-87fc-b372b34335e8">
It's not a View-related error. It's because the Course API is missing a building address ;> Please tell me if you ever encounter such error so I will update the building code. 

**Debugging building code file**
I have also added the missing building codes in codetoaddress.txt. To do so, I added a test file in debug_tool package to find me all the sessions with missing buildings code in the txt file. This text file takes forever to run given that we have 3000+ courses and we're using an external API. 
But this is the output the last time I ran it. The output shows that all courses codes should be correctly converted to Course entities. 
<img width="1042" alt="Screen Shot 2023-11-19 at 6 05 22 PM" src="https://github.com/52259tan/UofTCourseScheduleOptimizer/assets/144645042/c86dbe6b-3d95-4d07-9e32-d03b99b642a7">
